### ISSUE TITLE
Custom SDXL model not working 

### ISSUE THREAD

**Issue by Temerius02** [Acly/krita-ai-diffusion#323]
Hi, I want to say I love your Addon. 

Here is my problem. wanted to try out this specific Custom model
https://civitai.com/models/257749/pony-diffusion-v6-xl

I even spoke with the creator of this model, who said it should normally work with the normal SDXL comfiUI pipeline.

But when I try to use it produces pictures like this. 
![test](https://github.com/Acly/krita-ai-diffusion/assets/156337947/2a4ba8f6-2e3d-4c0c-b03d-b82c82fc2293)
Euler a with 25 steps and resolution of 1024px 

The model does use the SDXL VAE fix 
https://huggingface.co/madebyollin/sdxl-vae-fp16-fix

So it can't come from that. 

I do use the newest version of the Addon [Version 1.12.0] but I have used the addon since version [Version 1.9.0]

**Comment by AvidSpline** [comment]
I'm getting this too. It works fine with the base sdxl model and it works fine with the both that model and the other model (JuggernautXL) in comfyui. Just using the sdxl preset and switching the model. This happens in both live mode and regular gen mode, no errors in the console. I tried multiple samplers, I tried switching the VAE, I tried different resolutions and prompts. No loras or anything else. I also tried several different XL models and anything but the default seems to fail, although it could be sampling bias (I don't have many, these things are huge!).

**Comment by Danamir** [comment]
Can you try setting the "Clip Skip" to 2 in the corresponding style ? 

![image](https://github.com/Acly/krita-ai-diffusion/assets/15424198/b66c669d-0853-4083-87d6-ebfd2ba52aff)

NB : You can try to replicate the bug in ComfyUi by using a "CLIP Set Last Layer" node at -1 value. Some models works only without this node, or with at least a -2 value. Hence the corresponding setting in the plugin.

**Comment by Temerius02** [comment]
So it seems the problem was that the model didn't register as an SDXL derivative and as such didn't use the standard ClipSkip2 needed for an SDXL model

**Comment by Acly** [comment]
No clip skip 2 isn't standard. But I wasn't aware that explicitly adding the node with -1 messes up models that expect -2. We should just omit the node by default in that case.

**Comment by Danamir** [comment]
I concur, I tried a quick fix on my fork and it's working as expected : 

```python
    def clip_set_last_layer(self, clip: Output, clip_layer: int):
        if clip_layer == -1:
            return clip
        else:
            return self.add("CLIPSetLastLayer", 1, clip=clip, stop_at_clip_layer=clip_layer)
```

Do you want me to do a pull request ? It's only 3 lines of code so it may be quicker for you to add it directly to the main branch. 😅

**Comment by Acly** [comment]
Right, looks like I was wrong about clip 2 not being standard for SDXL, it is the default.

ComfyUI explicitly implements this for SDXL Clip, so I assume it makes sense, it also seems to have reasonable output if used with values beyond -2

Using -1 (last layer) probably bad for SDXL as it refers to special "hidden" layer...?

**Comment by Danamir** [comment]
I can confirm I was getting bad quality render with some very small prompts (ie. a single word) with clip skip 1, mostly on SDXL Turbo models. The results are much better with clip skip 2.

**Comment by Acly** [comment]
Fixed in [v1.13.0](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.13.0)
It doesn't change existing styles though which may still have an explicit clip skip of 1. Default and all new styles now have "default" clip last layer which doesn't insert the node at all.
